packages/form-control/src/types.ts

```diff
@@ -78,6 +85,20 @@ export interface Validator {
   isValid(instance: HTMLElement, value: FormValue): boolean;
 }
 
+export interface AsyncValidator extends ValidatorBase {
+   /**
+   * Callback for a given validator. Takes the FormControl instance
+   * and the form control value as arguments and returns a
+   * boolean to evaluate for that Validator as a promise.
+   * @param instance {FormControlInterface} - The FormControl instance
+   * @param value {FormValue} - The form control value
+   * @returns {Promise<boolean>} - The validity of a given Validator
+   */
+  isValid(instance: HTMLElement, value: FormValue, signal: AbortSignal): Promise<boolean|void>;
```

Comment:
why void return type? shouldnt this always return a boolean? is void only if the function rejects?

Reply:
If an async validator returns `Promise<void>`, then it has been cancelled or should have no effect on the validation state. This is for instances where a validation might want to be cancelled due to quick typing, etc.